analyses/cell-type-ewings/exploratory_analysis/08-merged-celltypes.Rmd

```diff
@@ -0,0 +1,694 @@
+
+Because we have some normal cell types that are showing gene signatures expected to be specific to tumor cells, we should use the AUC values to help pull out any remaining cells that should be tumor cells. 
+We can also use the up and down gene sets to distinguish EWS-high and EWS-low tumor cells. 
+
+Let's pick some gene sets where we see variation across cell types that we can use to define cutoffs: 
+
+- EWS-high gene sets: `aynaud-ews-targets` and `staege`
+- EWS-low gene sets: `wrenn-nt5e-genes` and `hallmark_EMT`
+
+```{r}
+genesets_of_interest = c(
+  "auc_aynaud-ews-targets",
+  "auc_staege",
+  "auc_wrenn-nt5e-genes",
+  "auc_hallmark_EMT"
+)
+
+cluster_density_plot(all_info_df, genesets_of_interest, "combined_lumped", "AUC")
+```
+
+We can use these plots to help define cutoffs of AUC values to use for re-labeling any normal cells that should be tumor cells. 
+We expect that the endothelial cells and macrophages have low AUC values so we can use these cell types as the background. 
+Any AUC values above that background should be classified as tumor. 
+
+With this in mind, let's use these cutoffs to define tumor cells:
+
+- `aynaud-ews-targets` > 0.025 AUC
+- `staege` > 0.025 AUC
+- `wrenn-nt5e-genes` > 0.1 AUC
+- `hallmark_EMT` > 0.05 AUC 
+
+Let's re-plot the density plots adding in a line for the AUC cutoffs we have chosen. 
+
+```{r}
+cutoffs = c(
+  "auc_aynaud-ews-targets" = 0.025,
+  "auc_staege" = 0.025,
+  "auc_wrenn-nt5e-genes" = 0.1,
+  "auc_hallmark_EMT" = 0.05
+)
+
+
+cutoffs |>
+    purrr::imap(\(cutoff, column){
+      plot_density(
+        all_info_df,
+        column,
+        "combined_lumped"
+      ) +
+        theme(text = element_text(size = 8)) +
+        geom_vline(xintercept = cutoff)
+    }) |>
+    patchwork::wrap_plots(ncol = 2)
+```
+
+I don't think these cutoffs are perfect, but I think they are reasonable. 
+Any cells that are to the left of these cutoffs will be classified as tumor. 
+
+We can also use these to further stratify tumor cells into EWS high and EWS low classes. 
+
+- `tumor EWS-low`: High expression of `wrenn-nt5e-genes` (> 0.1 AUC) and EMT markers `hallmark_EMT` (> 0.05 AUC)
+- `tumor EWS-high`: High expression of `aynaud-ews-targets` (>0.025 AUC) and `staege` (>0.025 AUC)
+
+```{r}
+# define "final.final" cell types 
+all_info_df <- all_info_df |> 
+  dplyr::mutate(
+    final_annotation = dplyr::case_when(
+      `auc_wrenn-nt5e-genes` > 0.1 & auc_hallmark_EMT > 0.05 ~ "tumor EWS-low",
+      `auc_aynaud-ews-targets` > 0.025 & auc_staege > 0.025 ~ "tumor EWS-high",
+      .default = consensus_annotation
+    ),
+    # lump together for easier plotting
+    final_lumped = final_annotation |>
+      forcats::fct_lump_n(9, other_level = "All remaining cell types", ties.method = "first") |>
+      forcats::fct_infreq() |>
+      forcats::fct_relevel("All remaining cell types", after = Inf)
+  )
```

Comment:
For the next plot, is it possible to keep the same order as the previous plot, just replace the single tumor row with the two tumor rows? I think that makes it a bit easier to compare these to the previous ones